### DISCUSSION TITLE
Can socket:// be used with nullmodem?

### DISCUSSION THREAD

**Discussion by alexrudd2** [pymodbus-dev/pymodbus#1865] [General]
`test/sub_examples/test_examples.py::TestAsyncExamples::test_async_simple_client[serial-rtu]`

creates `host == socket://__pymodbus_nullmodem:7636`.  Does this make any sense?

**Comment by janiversen** [comment]
why does it not make sense?

that is how to activate a nullmodem connection.

**Reply by alexrudd2** [reply]
OK, I would have expected `__pymodbus_nullmodem:7636` but clearly don't understand it fully :)

**Comment by janiversen** [comment]
NULLMODEM act as a host, so defining a serial connection, follow the same pattern as other hosts "socket://<host>:<port>".

**Comment by janiversen** [comment]
> OK, I would have expected `__pymodbus_nullmodem:7636` but clearly don't understand it fully :)

"socket://" signals to use a socket instead of a tty port for serial transport...no big secrets :-) and actually it does matter because this can be used by applications that use pymodbus.
